clients/venice-client/src/main/java/com/linkedin/venice/fastclient/meta/RequestBasedMetadata.java

```diff
@@ -152,7 +152,7 @@ private void discoverD2Service() {
    * @param onDemandRefresh
    * @return if the fetched metadata was an updated version
    */
-  private synchronized boolean updateCache(boolean onDemandRefresh) throws InterruptedException {
+  private synchronized void updateCache(boolean onDemandRefresh) throws InterruptedException {
     boolean updateComplete = true;
```

Comment:
It seems weird that `updateComplete` is `true` at the start of the function. Can we set it to `true` explicitly instead of marking it `false` when it fails?